### ISSUE TITLE
Deprecate usage of method based `@computed`.

### ISSUE THREAD

**Issue by rwjblue** [ember-decorators/ember-decorators#146]
Once https://github.com/rwjblue/ember-decorators/pull/141 lands, we should deprecate the usage of `@computed` with plain methods.

```js
class Foo extends Ember.Object { 
  // to be deprecated
  @computed
  derp() { }

  // new go forward suggestion
  @computed
  get derp() { }
}
```

Once that deprecation has been added and released, we can begin preparing for another major version bump to cull the deprecated functionality (this likely also means removing the dependency on `ember-macro-helpers`).

**Comment by buschtoens** [comment]
One thing that I really liked about the good ol' method based CP, was that they made it very easy / intuitive to follow the happy path of ["pure computed properties"](https://spin.atomicobject.com/2017/04/26/ember-pure-computed-properties/) (read "pure function") as you would get all values passed to your method.

This had the really nice side effect of not having to call `get(this, 'foo')` for each and every dependent key.

We won't have that with ES6 getters. I understand that the ultimate goal is to remove `Ember.get` and `Ember.set`, but during this transition phase, we'll have to write this awkward code again:

```js
class Foo extends Ember.Object { 
  @computed('foo', 'bar', 'quux')
  get derp() {
    const foo = get(this, 'foo');
    const bar = get(this, 'bar');
    const quux = get(this, 'quux');

    // or if you prefer the destructuring assignment

    const { foo, bar, quux } = this.getProperties('foo', 'bar', 'quux');

    return `${foo} ${bar} ${quux}`;
  }
}
```

**Comment by pzuraq** [comment]
We can definitely pull `ember-macro-helpers` into a new decorator library, maybe `ember-macro-decorators`. I think it should be its own repo so it can experiment with API (composeable computeds for instance is a whole other question) and focus on providing first class support for all the features of that library.

**Comment by buschtoens** [comment]
I think we should wait until we can release a first version of `ember-macro-helper-decorators` or something equivalent. So that users who rely on / prefer the "pure computed property" syntax / semantics (?) can just switch out the imports.

**Comment by pzuraq** [comment]
yeah, definitely agree. We shouldn't remove the current decorators until that happens, and it'll be a major version bump either way so I think we should iterate for a while longer before we do.

**Comment by rwjblue** [comment]
Yep, I definitely am not in a rush here.  Just notating the intent and trying to make sure we are all on the same page.

---

> This had the really nice side effect of not having to call get(this, 'foo') for each and every dependent key.

FWIW, I think the medium/longer term solution that I do quite like is:

```js
class Foo extends Ember.Object { 
  @computed('foo', 'bar', 'quux')
  get derp() {
    const { foo, bar, quux } = this;

    // or if you prefer the destructuring assignment

    const { foo, bar, quux } = this.getProperties('foo', 'bar', 'quux');

    return `${foo} ${bar} ${quux}`;
  }
}
```

Doing this requires some changes in Ember, but I think (generally speaking) most of Ember core is on board with this...

**Comment by pzuraq** [comment]
Fixed by #170
